### PR TITLE
248 feature add state exporter for supportissue

### PR DESCRIPTION
**Pull request by Neogeekmo**

### CODE REVIEW

**Comment by Kamian404** [comment]
@Neogeekmo Ready for code review

File: APR-Core/Config.lua [hunk 1]
```diff
@@ -181,6 +181,15 @@ function APR.settings:createBlizzOptions()
                     )
                 end
             },
+            statusButton = {
```
Comment (Neogeekmo) [inline APR-Core/Config.lua:184]: all buttons must be on the same line
so move it before resetButton and subtract the width of statusButton from the width of buttonOffset

Comment (Neogeekmo) [reply APR-Core/Config.lua:184]: image
image

buttonOffset = 0.75

File: APR-Core/StatusReport.lua [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+
+    return routeStep
+end
+
+
+function APR:getStatusReportInfos()
+    local coordinates
+    if not IsInInstance() then
+        local playerPosition = C_Map.GetPlayerMapPosition(C_Map.GetBestMapForUnit("player"),"player")
+        coordinates = math.ceil(playerPosition.x*10000)/100 .. " " .. math.ceil(playerPosition.y*10000)/100
```
Comment (Neogeekmo) [inline APR-Core/StatusReport.lua:66]: you can use APR.coordinate:RoundCoords(x, y, 2)

File: APR-Core/StatusReport.lua [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+		aprVersion = {"AddOn Version", APR.version},
+		wowVersion = {"Client Version", APR.wowpatch},
+		wowBuildVersion = {"Client Build", APR.wowbuild},
+		clientLanguage = {"Language", GetLocale()},
+		currentTime = {"Time & Date", date()},
+		currentRoute = {"Route", APR.ActiveRoute or NO_ACTIVE},
+		currentStep = {"Step", currentStep or NO_ACTIVE},
+		currentZone = {"Zone", GetRealZoneText() or UNKNOWN},
+		currentContinent = {"Continent", C_Map.GetMapInfo(APR:GetContinent()).name},
+		currentCoords = {"Coordinates", coordinates or INSTANCE},
```
Comment (Neogeekmo) [inline APR-Core/StatusReport.lua:80]: for me, missing World Coord since xe use it in our routes

File: APR-Core/Config.lua [hunk 4]
```diff
@@ -181,6 +181,15 @@ function APR.settings:createBlizzOptions()
...
+                order = 1.5,
+                name = "Status",
```
Comment (Neogeekmo) [inline APR-Core/Config.lua:186]: don't we put this in CF for translation and just not translate the frame content ?.
but do it for the buttons?

File: APR-Core/StatusReport.lua [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+	LogoTop:SetPoint('CENTER', titleLogoFrame, 'TOP', 0, -36)
+	LogoTop:SetTexture("Interface\\AddOns\\APR-Core\\assets\\APR_logo")
+	LogoTop:SetSize(64, 64)
+	titleLogoFrame.LogoTop = LogoTop
+
+	--CopyButton to export the infos - TODO Styling
+    local CopyButton = CreateFrame('Button', nil, StatusFrame, "StaticPopupButtonTemplate")
+    CopyButton:SetPoint("BOTTOM", 0, 5)
+    CopyButton:SetSize(100, 20)
+    CopyButton:SetText("Export Information")
```
Comment (Neogeekmo) [inline APR-Core/StatusReport.lua:209]: same translation ?

File: APR-Core/StatusReport.lua [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+	local CopyButtonFont = CopyButton:CreateFontString()
+	CopyButtonFont:SetFont(font, 9)
+	CopyButtonFont:SetText("Export Information")
```
Comment (Neogeekmo) [inline APR-Core/StatusReport.lua:212]: same translation ?

File: APR-Core/StatusReport.lua [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+
+function APR:createStatusStaticContent(StatusFrame)
+	local statusInfos = APR:getStatusReportInfos()
+
+	--Section 1 AddOn & WoW Info
+	StatusFrame.Section1 = APR:createStatusSection(300, 90, nil, 30, StatusFrame, 'TOP', StatusFrame, 'TOP', -30)
+	StatusFrame.Section1.Content = APR:createStatusContent(4, 260, StatusFrame.Section1, StatusFrame.Section1.Header)
+	StatusFrame.Section1.Header.Text:SetFormattedText('Addon & Client|r')
+
+	StatusFrame.Section1.Content.Line1.Text:SetFormattedText(statusInfos.aprVersion[1] .. ': |cff' .. APR.HEXColor.green .. '%s|r', statusInfos.aprVersion[2]) --ToDo: Check if Addon outdated
```
Comment (Neogeekmo) [inline APR-Core/StatusReport.lua:230]: create an other issue for this TODO

File: APR-Core/StatusReport.lua [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+function APR:updateStatusFrame()
+	local StatusFrame = APR.StatusFrame
+	local statusInfos = APR:getStatusReportInfos()
+    local statusColors = APR:getStatusColors()
+
+	StatusFrame.Section1.Content.Line4.Text:SetFormattedText(statusInfos.currentTime[1] .. ': |cff' .. APR.HEXColor.green ..'%s|r', statusInfos.currentTime[2])
+
+	StatusFrame.Section2.Content.Line1.Text:SetFormattedText(statusInfos.currentRoute[1] .. ': |cff' .. statusColors.currentRouteColor ..'%s|r', statusInfos.currentRoute[2])
+	StatusFrame.Section2.Content.Line2.Text:SetFormattedText(statusInfos.currentStep[1] .. ': |cff' .. statusColors.currentStepColor ..'%s|r', statusInfos.currentStep[2])
+	StatusFrame.Section2.Content.Line3.Text:SetFormattedText(statusInfos.currentZone[1] .. ': |cff' .. statusColors.currentZoneColor ..'%s|r', statusInfos.currentZone[2])
```
Comment (Neogeekmo) [inline APR-Core/StatusReport.lua:258]: put Zone and Continent on the same line with this format : Zone, Continent

File: APR-Core/StatusReport.lua [hunk 9]
```diff
@@ -0,0 +1,290 @@
...
+	StatusFrame.Section2.Content.Line4.Text:SetFormattedText(statusInfos.currentContinent[1] .. ': |cff4beb2c%s|r', statusInfos.currentContinent[2])
+	StatusFrame.Section2.Content.Line5.Text:SetFormattedText(statusInfos.currentCoords[1] .. ': |cff' .. statusColors.currentCoordsColor .. '%s|r', statusInfos.currentCoords[2])
```
Comment (Neogeekmo) [inline APR-Core/StatusReport.lua:260]: maybe add the world coord in parenthesis after the map coordinate (you can use UnitPosition("player") + APR.coordinate:RoundCoords(x, y, 2) or math.ceil(x*10000)/100

File: APR-Core/StatusReport.lua [hunk 10]
```diff
@@ -0,0 +1,290 @@
...
+
+	StatusFrame.Section3.Content.Line3.Text:SetFormattedText(statusInfos.charLevel[1] .. ': |cff' .. APR.HEXColor.green .. '%s|r', statusInfos.charLevel[2])
+end
+
+function APR:getCurrentRouteInfos()
+	local pos = C_Map.GetPlayerMapPosition(C_Map.GetBestMapForUnit("player"),"player")
+	pos = math.ceil(pos.x*10000)/100 .. " " .. math.ceil(pos.y*10000)/100
```
Comment (Neogeekmo) [inline APR-Core/StatusReport.lua:267]: same for coord fonction

**Comment by Neogeekmo** [comment]
@Kamian404 :)

**Review by Kamian404 — APPROVED** [review]

**Pull request merged**